[Idea/review] map[string]interface{} as internal data for changeset

pijalu opened the pull request:
Here comes a PR to change internal data of changeset from "json" to a more generic map[string]interface{}

The new parser packages are in charge to glue sources and changesets.

It probably needs a bit more work/testing, but I wonder if I should continue in that direction or not :)

asim commented:
It's an interesting idea but one that needs to be thought about a little bit more. Returning `map[string]interface{}` as the data type does make sense but I need to look at the sources a bit more to understand it. I think the idea behind returning bytes was so the source doesn't have to interpret the data but that's hard in certain places.

Changeset should remain in the source package. Value should remain in the reader package. Parser is a useful new package.

pijalu commented:
No worries !
I did it for as an experiment/for the lulz/too much nice weather during the WE ;-) !

Some justifications/what I got playing around to have the ideas flowing: 
* I tried first to pass/use a "format" flag but I always ended with the source setting the format in some ways. Even if it's just to pass a file name, it already know too much
* Options looked overkill as I would curse libs that ask me to confirm the format... when it's part of the name
* Lots of sources have a single format and I didn't like the internal is JSON and link with 2 json libs when every sources was handling map[string]interface{} ==> map[string]interface looked like a good candidate.

Side note: Updated PR: I moved back changeset to source - Value wasn't moved AFAIK

Cheers !

asim commented:
This was good inspiration, thanks for the work. Opted to go down a similar path but slightly different.
